File: ecommerc_dotnet/dto/Request/ProductVarientRequestDto.cs
using ecommerc_dotnet.Validation;

namespace ecommerc_dotnet.dto.Request;

public class ProductVarientRequestDto
{
    [StringNotNullOrEmptyValidation(ErrorMessage= "name must not be empty")]
    public string? name { get; set; }
    public decimal? precentage { get; set; } = 0;
    public Guid varient_id { get; set; }
}

File: eccorce_app/app/src/main/java/com/example/eccomerce_app/dto/request/ProductVarientRequestDto.kt
package com.example.e_commercompose.dto.request

import androidx.compose.runtime.Composable
import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class ProductVarientRequestDto(
    val name: String,
    val precentage: Double?,
    @Serializable(with= UUIDKserialize::class)
    val varient_id: UUID,
    )
